packages/@n8n/db/src/repositories/execution.repository.ts

```diff
@@ -960,8 +961,14 @@ export class ExecutionRepository extends Repository<ExecutionEntity> {
 
 		const qb = this.createQueryBuilder('execution')
 			.select(fields)
-			.innerJoin('execution.workflow', 'workflow')
-			.where('execution.workflowId IN (:...accessibleWorkflowIds)', { accessibleWorkflowIds });
+			.innerJoin('execution.workflow', 'workflow');
+
+		// Add join to execution_data only if nodesExecuted is not null or not an empty array
+		if (nodesExecuted && Array.isArray(nodesExecuted) ? nodesExecuted.length > 0 : true) {
```

Comment:
Won't this always be true? 
If nodesExecuted is undefined?, then `true` is returned, 
If it's an array, `nodesExecuted.length > 0` is returned which is also true